src/vkscript/command_parser.cc

```diff
@@ -633,10 +633,6 @@ Result CommandParser::ProcessSSBO() {
     if (value_count > buf->ValueCount())
       buf->SetValueCount(value_count);
 
-    // Even if the value count doesn't change, the buffer is still resized
-    // because this maybe the first time data is set into the buffer.
-    buf->ResizeTo(buf->GetSizeInBytes());
```

Comment:
This is required, please don't remove this.

Reply:
The problem is for the buffers with pack of zero each time we process ssbo,
it resizes the buffer. As a result even if the valueCount is the same the
size gets multiplied by ElementCount().
However the buffer is resized correctly whenever we call
setDataWithOffset().

Reply:
Dan: 
'm not sure what you mean by a pack of zero? The reason we have to do this is because we may have an ssbo size, but never set any data into the buffer. In the case of an ssbo size we have to use the number as the byte size as we don't know the size of the element, so we assume each element is 1 byte.

So, you do:

ssbo size foo 100
draw ...
probe foo 100 2 2
The draw shader, uses the ssbo as output, so it has to have the size set, but we never set any data. So, we don't really know what's in the buffer, and we have to use the 100 as the byte size.

Reply:
The outcome of not removing this line is the buffer size getting bigger at each processSSBO() call. For example when I run something like after the first call buffer size is 256 and the next time is 4096.

ssbo 0 subdata vec4 0 1.0 0.0 0.0 1.0
ssbo 0 subdata vec4 0 0.0 1.0 0.0 1.0

I'm looking in the code to why is this happening (yesterday I found it: an if condition base on format's pack size but I cannot find it right now, I don't know why :( ).
-----------------------
I don't think this affects "ssbo size" since it is only executed when the token is "ssbo subdata". Also I added this line and a couple lines before it to set the max ssbo size, it wasn't in the code previously. 
----------------------
I keep looking to see why each call to resize is making the buffer bigger although the elementCount is not changed.

Reply:
In the above example, it may call the resize, but it should do nothing as the size didn't change with the `subdata` call? If it's resizing something is wrong.

if you had:

```
ssbo size 0:0 400
ssbo 0:0 subdata vec4 0 0 1 0 0
```

the ssbo should still be `400` afterwards as it's the larger size, resize should only go up.

Reply:
Ohh, @sarahM0 from your example, is the complete example something like:

```
ssbo size 256
ssbo subdata vec4 0 1 2 3 4
```

? If so, I'm guessing the bug is that the `ssbo size` set the element count to `256`. We then changed from a format of `byte` to `vec4` which has a much larger memory size. We originally had `256 * byte` size and changed it to `256 * 4 * 4`?

If so, I think you'll need to track in the buffer if we had just a `size` command, and don't take that as the element count if we're re-calculating the size after a format change. Or .... something along those lines.

Reply:
Let's say we have a simple script like this:
---------------------------------------------------------------------------------
ssbo 0 subdata vec4 0 1.0 0.0 0.0 1.0
draw rect -1 -1 1 1
relative probe rect rgba (0.0, 0.0, 0.5, 0.5) (1.0,   0,   0, 1.0)

ssbo 0 subdata vec4 0 0.0 1.0 0.0 1.0
draw arrays TRIANGLE_LIST 0 6
relative probe rect rgb (0.0, 0.0, 1.0, 1.0) (0, 1.0, 
---------------------------------------------------------------------------------
It has to do with the ValueCount() function in buffer.h:
  /// Returns the number of values in the buffer.
  uint32_t ValueCount() const {
    if (!format_)
      return 0;
    // Packed formats are single values.
    if (format_->GetPackSize() > 0)
      return element_count_;
    return element_count_ * format_->ValuesPerElement();
  }

When pack_size_in_bytes_ of the buffer format object is 0 the return value is multiplied by format_->ValuesPerElement().
pack_size_in_bytes_ == 0 for most formats, those the do not have PACK in them, such as UNORM and UFLOAT.

Let me know what you think. I'll revert my changes in commad_parser.cc and open another PR for this.

Reply:
RIght, but in that script, in both cases element count should be 1 and valuesPerElement should be 4. So, the size should be correct?

This is only a problem with using ssbo size before subdata, yea?

Reply:
(and, yes, please file a bug for this so we can figure out what's going wrong and the combination of things that cause it)

Reply:
You're right the first time:  element count == 1 and valuesPerElement == 4
Then we do:
    if (value_count > buf->ValueCount())
      buf->SetValueCount(value_count);
as a result: element count == 4. Then we do:
    buf->ResizeTo(buf->GetSizeInBytes());
Hence the size increases at each call.

Reply:
Issue #575